scaffold-eth tutorial added (#25)
kaiachain/kaia-docs: docs/build/tutorials/scaffold-eth.md

# Build a dApp using Scaffold-ETH 2

## Introduction <a href="#introduction" id="introduction"></a>

Scaffold-ETH 2 is an open-source toolkit for building decentralized applications (dApps) on Ethereum and other EVM-compatible blockchains, like Kaia. Developers can easily deploy a Solidity smart contract and launch a dApp with a React frontend thanks to Scaffold-ETH 2.

The Scaffold-ETH 2 toolkit was built using Next.js, RainbowKit, Hardhat, Foundry, Wagmi, and TypeScript. Developers can easily create, test, and deploy smart contracts using Hardhat or Foundry, as well as build a React frontend using Next.js.

In this tutorial, you will learn how to deploy, run a contract and build a dApp on Kaia using Scaffold-ETH 2.

## Prerequisites <a href="#prerequisites" id="prerequisites"></a>

> To get started in this guide, you will need:

Reviewer: ```suggestion
To get started with this guide, you will need:
```

**Step 1: Create .env**

To create .env file, copy and paste the code below in your terminal

```bash
touch packages/hardhat/.env
```

You can refer to the **.env.example** file for the variables that are already used in the hardhat.config.js file. For Kaia, you'll only need to create one variable: **DEPLOYED_PRIVATE_KEY**.

**Step 2: Edit your .env file to include this variable:**

```bash
DEPLOYER_PRIVATE_KEY=INSERT_PRIVATE_KEY
```

The private key stated in your **.env** file corresponds to the account that will deploy and interact with the smart contracts in your Hardhat project.

**Step 3: Modify hardhat.config.ts**

The next thing we want to do is to configure **hardhat.config.ts** to support Kaia.

Set the constant **defaultNetwork** to the network you are deploying the smart contract to. 

```js
defaultNetwork = "kaia"
```

Add the network configurations for Kaia under the networks configuration object

```js
kaia: {
  url: "INSERT_RPC_URL",
  accounts: [deployerPrivateKey],
},
```

For more information on using Hardhat with Kaia, please check [Hardhat guide](https://docs.kaia.io/docs/build/get-started/hardhat/) for more details.

### Deploy Contract to Kaia

> Having configured hardhat to support Kaia network, the next thing we want to do is to compile and deploy the sample contract. 

Reviewer: ```suggestion
After configuring Hardhat to support the Kaia network, the next step is to compile and deploy the sample contract.
```

![Verify](/img/build/tutorials/sc-verify.png)
 
![Verify on Klaytnscope](/img/build/tutorials/sc-verify-klaytnscope.png)

For more information about verifying smart contracts on Kaia using the Hardhat Verify plugin, please refer to the H[ardhat-Verify-Plugins guide](https://docs.kaia.io/docs/build/smart-contracts/verify/hardhat/).

## Next.js Configuration <a href="#nextjs-configuration" id="nextjs-configuration"></a>

In this section, you'll modify the Next.js configuration to target Kairos Testnet (where the smart contract was deployed to) under the **packages/nextjs** folder. In this folder, we intend to modify the **targetNetwork** array in the scaffoldConfig object in **scaffold.config.ts** file. 

### Modify the targetNetwork array <a href="#modify-targetnetwork-array" id="modify-targetnetwork-array"></a>

```js
targetNetworks: [chains.klaytnBaobab],
```

That's all required to configure Next.js! Next, is to launch the dApp in your localhost. 

### Launch the dApp in your Localhost <a href="#launch-dapp-in-localhost" id="launch-dapp-in-localhost"></a>

> Having made all necessary configurations, you can now launch the example dApp in your localhost. 

Reviewer: ```suggestion
After making all the necessary configurations, you can now launch the example dApp on your localhost.
```

To do so, run:

```bash
yarn start
```

![Run dApp](/img/build/tutorials/sc-run-dapp.png)

You should now be able to access a React-based dApp frontend at http://localhost:3000/. Feel free to  interact with the dApp by connecting your wallet or checking out the contract debugger page.

![Scaffold dApp](/img/build/tutorials/sc-dapp.png)

## Conclusion

> Congratulations on using the Scaffold-ETH 2 to deploy and run a contract and a dApp on Kaia. Now that you understand the workings of Scaffold-ETH 2, feel free to create and deploy your own smart contracts and modify the frontend to fit your dApp's needs!

Reviewer: ```suggestion
Congratulations! You have successfully used Scaffold-ETH 2 to deploy a contract and run a dApp on Kaia. Now that you understand the workings of Scaffold-ETH 2, feel free to create and deploy your own smart contracts and modify the frontend to fit your dApp's needs!
```
